integration-tests/src/test/java/oracle/kubernetes/operator/utils/DomainCRD.java

```diff
@@ -0,0 +1,200 @@
+      ((ObjectNode) adminServerNode).put(entry.getKey(), entry.getValue());
+    }
+  }
+
+  /**
+   * A utility method to add attributes to cluster node in domain.yaml
+   *
+   * @param ClusterName - Name of the cluster to which the attributes to be added
+   * @param attributes - A HashMap of key value pairs
+   */
+  public void addObjectNodeToCluster(String ClusterName, Map<String, String> attributes) {
+
+    JsonNode clusterNode = getClusterNode(ClusterName);
+    for (Map.Entry<String, String> entry : attributes.entrySet()) {
+      ((ObjectNode) clusterNode).put(entry.getKey(), entry.getValue());
+    }
+  }
+
+  /**
+   * A utility method to add attributes to managed server node in domain.yaml
+   *
+   * @param managedServerName - Name of the managed server to which the attributes to be added
+   * @param attributes - A HashMap of key value pairs
+   */
+  public void addObjectNodeToMS(String managedServerName, Map<String, String> attributes) {
+    JsonNode managedServerNode = getManagedServerNode(managedServerName);
+    for (Map.Entry<String, String> entry : attributes.entrySet()) {
+      ((ObjectNode) managedServerNode).put(entry.getKey(), entry.getValue());
+    }
+  }
+
+  /** Gets the spec node entry from Domain CRD JSON tree */
+  private JsonNode getSpecNode() {
+    return root.path("spec");
+  }
+
+  /**
+   * Gets the administration server node entry from Domain CRD JSON tree
+   *
+   * @return
+   */
+  private JsonNode getAdminServerNode() {
+    return root.path("spec").path("adminServer");
+  }
+
+  /**
+   * Gets the cluster node entry from Domain CRD JSON tree for the given cluster name
+   *
+   * @param root - Root JSON node of the Domain CRD JSON tree
+   * @param clusterName - Name of the cluster
+   * @return - cluster node entry from Domain CRD JSON tree
+   */
+  /**
+   * Gets the cluster node entry from Domain CRD JSON tree for the given cluster name
+   *
+   * @param clusterName - Name of the cluster
+   * @return - JsonNode of named cluster
+   */
+  private JsonNode getClusterNode(String clusterName) {
+    ArrayNode clusters = (ArrayNode) root.path("spec").path("clusters");
+    JsonNode clusterNode = null;
+    for (JsonNode cluster : clusters) {
+      if (cluster.get("clusterName").asText().equals(clusterName)) {
+        System.out.println("Got the cluster");
+        clusterNode = cluster;
+      }
+    }
+    return clusterNode;
+  }
+
+  /**
+   * Gets the managed server node entry from Domain CRD JSON tree
+   *
+   * @param managedServerName Name of the managed server for which to get the JSON node
+   * @return managed server node entry from Domain CRD JSON tree
+   */
+  private JsonNode getManagedServerNode(String managedServerName) {
+    ArrayNode managedservers = null;
+    JsonNode managedserverNode = null;
+    if (root.path("spec").path("managedServers").isMissingNode()) {
+      System.out.println("Missing MS Node");
```

Comment:
use logger instead of s.o.p in all places..

Reply:
Fixed